ta4j-core/src/main/java/org/ta4j/core/BaseBar.java

```diff
@@ -90,6 +92,47 @@ public BaseBar(ZonedDateTime endTime, double openPrice, double highPrice, double
                 numFunction.apply(closePrice),
                 numFunction.apply(volume),numFunction.apply(0));
     }
+    
+    /**
+     * Constructor.
+     * @param timePeriod the time period
+     * @param endTime the end time of the bar period
+     * @param openPrice the open price of the bar period
+     * @param highPrice the highest price of the bar period
+     * @param lowPrice the lowest price of the bar period
+     * @param closePrice the close price of the bar period
+     * @param volume the volume of the bar period
+     * @param numFunction the numbers precision
+     */
+    public BaseBar(Duration timePeriod, ZonedDateTime endTime, BigDecimal openPrice, BigDecimal highPrice, BigDecimal lowPrice, BigDecimal closePrice, BigDecimal volume, Function<Number, Num> numFunction) {
+    	this(timePeriod, endTime, openPrice, highPrice, lowPrice, closePrice, volume, numFunction.apply(0), numFunction);
```

Comment:
Is it necessary to provide a `numFunction` parameter? Or can we use `PrecisionNum` here so that you do not have to create an own `Function<Number, Num>`?

Reply:
yes, good idea. I changed constructor so "PrecisionNum.valueOf()" is used for BigDecimal.